cmd/flexvolume/cmd/unmount.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2017 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/volume/util"
+
+	"github.com/rook/rook/pkg/agent/flexvolume"
+	"github.com/spf13/cobra"
+)
+
+var (
+	unmountCmd = &cobra.Command{
+		Use:   "unmount",
+		Short: "Unmounts the pod volume",
+		RunE:  unmount,
+	}
+)
+
+func init() {
+	RootCmd.AddCommand(unmountCmd)
+}
+
+func unmount(cmd *cobra.Command, args []string) error {
+
+	client, err := getRPClient()
+	if err != nil {
+		return fmt.Errorf("Rook: Error getting RPC client: %v", err)
+	}
+
+	// unmounts the volume
+	mountDir := args[0]
+	log(client, fmt.Sprintf("unmounting mount dir: %s", mountDir), false)
+
+	mounter := getMounter()
+
+	err = redirectStdout(
+		client,
+		func() error {
+			if err := util.UnmountPath(mountDir, mounter); err != nil {
+				return fmt.Errorf("failed to unmount volume at %s: %+v", mountDir, err)
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	var opts = &flexvolume.AttachOptions{
+		MountDir: mountDir,
+	}
+	// call detach
+	log(client, "calling agent to detach volume ", false)
```

Comment:
I cant print the volume name because it is not passed by kubernetes. As for the mountDir, im already logging it on line 49. But ill also print it here as well.